### PR TITLE
General code improvements (LibHTTP, LibURL, Fuzzers)

### PR DESCRIPTION
**Pull request by lalitshankarchowdhury**
- LibHTTP: Enforced proper typecasting and added const correctness
- LibURL: Fixed a case of `ignore0` being checked before assignment
- Fuzzers: Fixed some memory leaks

### CODE REVIEW

File: Userland/Libraries/LibURL/Parser.cpp [hunk 1]
```diff
@@ -304,9 +304,8 @@ static void serialize_ipv6_address(IPv6Address const& address, StringBuilder& ou
         if (ignore0 && address[piece_index] == 0)
             continue;
 
-        // 2. Otherwise, if ignore0 is true, set ignore0 to false.
-        if (ignore0)
-            ignore0 = false;
+        // 2. Set ignore0 to false.
+        ignore0 = false;
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibURL/Parser.cpp:308]: The comment was a direct quote from the spec. Please don't change it.

File: Userland/Libraries/LibHTTP/Job.cpp [hunk 2]
```diff
@@ -590,7 +590,7 @@ void Job::finish_up()
         auto flattened_buffer = maybe_flattened_buffer.release_value();
 
         u8* flat_ptr = flattened_buffer.data();
-        for (auto& received_buffer : m_received_buffers) {
+        for (const auto& received_buffer : m_received_buffers) {
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibHTTP/Job.cpp:593]: We use east const. eg, `auto const&`

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
Hi there, welcome to Ladybird. :^)

Please try to give descriptive names to your commits. "LibHTTP: Type fixes" doesn't really say anything, and "LibURL: Refactor" especially doesn't.

I'm going to run CI just to catch any more formatting issues.

File: Userland/Libraries/LibURL/Parser.cpp (hunk 1 above)
Comment (trflynn89) [reply Userland/Libraries/LibURL/Parser.cpp:308]: > LibURL: Fixed a case of ignore0 being checked before assignment

That isn't the case here though? It's initialized just a couple lines above here.
```c++
// 4. Let ignore0 be false.
auto ignore0 = false;
```

Comment (lalitshankarchowdhury) [reply Userland/Libraries/LibURL/Parser.cpp:308]: > > LibURL: Fixed a case of ignore0 being checked before assignment
> 
> That isn't the case here though? It's initialized just a couple lines above here.
> 
> ```c++
> // 4. Let ignore0 be false.
> auto ignore0 = false;
> ```

Please mind that I am just removing the redundant check for `ignore0`.

**Pull request closed without merging**